component-classes/index.js

```diff
@@ -398,13 +398,14 @@ export const input = {
 };
 
 export const select = {
-  default: 'block text-m mb-0 leading-m s-text s-bg s-border hover:s-border-hover active:s-border-active rounded-4 py-12 px-8 block border-1 w-full focusable focus:[--w-outline-offset:-2px] appearance-none pr-32 cursor-pointer caret-current',
-  disabled: 's-bg-disabled-subtle s-border-disabled hover:s-border-disabled! active:s-border-disabled! s-text-disabled pointer-events-none',
-  invalid: 's-border-negative hover:s-border-negative-hover! outline-[--w-s-color-border-negative]!',
-  readOnly: 'pl-0 bg-transparent border-0 pointer-events-none before:hidden',
+  base: 'block text-m mb-0 py-12 pr-32 rounded-4 w-full focus:[--w-outline-offset:-2px] appearance-none cursor-pointer caret-current',
+  default: 's-text s-bg pl-8 border-1 s-border hover:s-border-hover active:s-border-active focusable',
+  disabled: ' s-text-disabled s-bg-disabled-subtle pl-8 border-1 s-border-disabled hover:s-border-disabled active:s-border-disabled pointer-events-none focusable',
+  invalid: 's-text s-bg pl-8 border-1 s-border-negative hover:s-border-negative-hover active:s-border-active outline-[--w-s-color-border-negative]',
```

Comment:
Is disabled state supposed to be focusable while invalid is not?

Reply:
I think it should. From MDN:
["Read-only elements are still relevant to the user, so you should not prevent the user from navigating to the element or its focusable descendants or copying the value."](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-readonly#:~:text=Read%2Donly%20elements%20are%20still%20relevant%20to%20the%20user%2C%20so%20you%20should%20not%20prevent%20the%20user%20from%20navigating%20to%20the%20element%20or%20its%20focusable%20descendants%20or%20copying%20the%20value.)

Reply:
So before the changes, `disabled`, `invalid` and `readOnly` got `focusable` from the `default` that sets outline to none. While invalid had also a `outline-[--w-s-color-border-negative]!`, this overwrote focusable.

Reply:
Ah alright! I'll implement the same for select then, as you did for input! :) 👍

Reply:
As for disabled, the `focusable` styles have no effect because `disabled` attribute of `select` element makes it non-interactive. You could then move the `focusable` class to the `select.base`, just like it was done in `input` classes, because in such case all other variants get the focus styling they expect, while it won't have any effect on the disabled variant.

Reply:
Updated in this commit: [a82dd29](https://github.com/warp-ds/css/pull/213/commits/a82dd29b799369cd46a5c2774bbeecb50d797d0f) :)